File: frontend/src/app/core/services/review.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { Review, ReviewRequest } from '../../shared/models/booking.model';

@Injectable({ providedIn: 'root' })
export class ReviewService {
  private api = inject(ApiService);

  submit(req: ReviewRequest): Observable<Review> {
    return this.api.post<Review>('/reviews', req);
  }

  getByTutor(tutorUserId: number): Observable<Review[]> {
    return this.api.get<Review[]>(`/reviews/tutor/${tutorUserId}`);
  }

  hasReviewed(bookingId: number): Observable<boolean> {
    return this.api.get<boolean>(`/reviews/check/${bookingId}`);
  }

  addReply(reviewId: number, reply: string): Observable<Review> {
    return this.api.patch<Review>(`/reviews/${reviewId}/reply`, { reply });
  }
}

File: backend/src/main/java/com/tutorfinder/service/ReviewService.java
package com.tutorfinder.service;

import com.tutorfinder.dto.ReviewDto;
import com.tutorfinder.dto.ReviewRequest;
import java.util.List;

public interface ReviewService {
    ReviewDto submitReview(Long studentId, ReviewRequest req);
    List<ReviewDto> getReviewsByTutor(Long tutorUserId);
    ReviewDto addTutorReply(Long reviewId, Long tutorUserId, String reply);
    boolean hasReviewed(Long bookingId);
}
